sev/src/certs/crypto.rs

```diff
@@ -110,22 +110,34 @@ where
             return Err(ErrorKind::InvalidInput.into());
         }
 
-        let mut ver = sign::Verifier::new(sig.hash, &self.key)?;
-        if self.key.id() == pkey::Id::RSA {
-            ver.set_rsa_padding(rsa::Padding::PKCS1_PSS)?;
-            ver.set_rsa_pss_saltlen(sign::RsaPssSaltlen::DIGEST_LENGTH)?;
+        let mut verifiers = [
+            // Common case
+            sign::Verifier::new(sig.hash, &self.key)?,
+            // Rome CA certificate quirk
+            // The Naples ARK/ASK use a SHA256 digest whereas the Rome
+            // ones use a SHA384 digest. These certificates do not explicitly
+            // advertise which one we should use here, and there is no convenient
+            // way to deduce which kind of certificate we have here except to see
+            // if we can successfully verify the certificate with SHA384.
+            sign::Verifier::new(openssl::hash::MessageDigest::sha384(), &self.key)?,
+        ];
```

Comment:
This is, unfortunately, precisely what we **MUST NOT** do. You've now just given an attacker two different ways to get a false positive for every certificate.

Instead, you need to change the values [here](https://github.com/enarx/enarx/blob/master/sev/src/certs/ca/cert/v1.rs#L235) and [here](https://github.com/enarx/enarx/blob/master/sev/src/certs/ca/cert/v1.rs#L260) based on the **size** of the RSA key in the certificate. 2k RSA always uses SHA256 and 4k RSA always uses SHA384.